source/gui/addonStoreGui/controls/actions.py

```diff
@@ -154,6 +154,13 @@ def _actions(self) -> List[BatchAddonActionVM]:
 				validCheck=lambda aVMs: AddonListValidator(aVMs).canUseUpdateAction(),
 				actionTarget=self._selectedAddons,
 			),
+			BatchAddonActionVM(
+				# Translators: Label for an action that retries the selected add-ons
+				displayName=pgettext("addonStore", "Re&try installing selected add-ons"),
```

Comment:
English wording suggestions; to be confirmed by a native English speaker before ecceptation (cc @seanbudd, @XLTechie)

Since the action for a single add-on is "Retry install", implied "Retry 'Install' action", I'd say that the batch action should be "Retry install selected add-ons", implied "Retry the action 'Install the selected add-ons'" (beware to quotes!).

```suggestion
				displayName=pgettext("addonStore", "Re&try install selected add-ons"),
```

Or if you want to keep "installing", should we also use it for the simple action?

But in this case, rather than "installing", I would have expected "to install".

```suggestion
				displayName=pgettext("addonStore", "Re&try to install selected add-ons"),
```

Alternatively, another shorter wording:
```suggestion
				displayName=pgettext("addonStore", "Install selected add-ons (re&try)"),
```

and "Install (re&try)" for the single add-on.

Reply:
> Or, let the operation fail, tell the user it failed, and restore the menu to the pre-failure state and let the user start again as if nothing had been attempted, with no "Retry" anything.

This changes the original state change logic.

Of course the smallest change is just to directly re-determine the status of the add-on when the error dialog pops up.

Some status display strings are no longer used.

Reply:
>Or, let the operation fail, tell the user it failed, and restore the menu to the pre-failure state and let the user start again as if nothing had been attempted, with no "Retry" anything.

Actually, I would say this is close to the current behaviour in NVDA. you get the status, change tabs or re-open the store, try again.

Reply:
@CyrilleB79 - I don't think any of these suggestion are valid english grammar except the last one, but I think the current wording is more consistent with other batch actions